apps/docs/docs/react/make-styles.md

```diff
@@ -0,0 +1,192 @@
+---
+sidebar_position: 2
+---
+
+# makeStyles
+
+Is used to define styles, returns [a React hook](https://reactjs.org/docs/hooks-intro.html) that should be called inside a component:
+
+```jsx
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  button: { color: 'red' },
+  icon: { paddingLeft: '5px' },
+});
+
+function Component() {
+  const classes = useClasses();
+
+  return (
+    <div>
+      <button className={classes.button} />
+      <span className={classes.icon} />
+    </div>
+  );
+}
+```
+
+### Nesting selector
+
+:::caution Use with caution
+
+One of the promises of Atomic CSS is reducing the amount of styles, however more selectors are created when nested.
+
+:::
+
+The `&` character references the parent selector(s):
+
+```js
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  root: {
+    '& .foo': { color: 'green' },
+    '&.bar': { color: 'red' },
+  },
+});
+```
+
+Produces following CSS:
+
+```css
+.f15f830o .foo {
+  color: green;
+}
+.fns74iw.bar {
+  color: red;
+}
+```
+
+:::tip
+
+As in CSS spacing between `&` and a selector matters, [What’s the Difference?](https://css-tricks.com/whats-the-difference/)
+
+:::
+
+### Pseudo-classes
+
+These rules are transformed to receive a nesting selector:
+
+```js
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  root: {
+    ':active': { color: 'pink' },
+    ':hover': { color: 'blue' },
+    // 💡 :link, :focus, etc. are also supported
+
+    ':nth-child(2n)': { backgroundColor: '#fafafa' },
+  },
+});
+```
+
+```css
+.fqsesyh:active {
+  color: pink;
+}
+.f10q6zxg:hover {
+  color: blue;
+}
+.fnbrw4x:nth-child(2n) {
+  background-color: #fafafa;
+}
+```
+
+### `:global()` selector
+
+Another useful feature is `:global()` selector, it allows connecting local styles with global selectors.
+
+```ts
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  root: {
+    ':global(html[data-whatintent="mouse"])': { backgroundColor: 'yellow' },
+  },
+});
+```
+
+```css
+html[data-whatintent='mouse'] .f1xz3i88 {
+  background-color: yellow;
+}
+```
+
+### At-rules
+
+`@media` & `@supports` queries are also supported:
```

Comment:
nit: should we have example output for @media and @support query example you have here ?

Reply:
Yeah, will add it 👍